bedwars-plugin/src/main/java/com/tomkeuper/bedwars/arena/tasks/GamePlayingTask.java

```diff
@@ -175,36 +176,37 @@ public void run() {
-                            distance = (int) p.getLocation().distance(p2.getLocation());
-                        } else if ((int) p.getLocation().distance(p2.getLocation()) < distance) {
-                            distance = (int) p.getLocation().distance(p2.getLocation());
+        if (config.getBoolean(ConfigPath.GENERAL_CONFIGURATION_ENABLE_TEAMMATE_TRACKING_ACTION_BAR)) {
+            int distance = 0;
+            for (ITeam t : getArena().getTeams()) {
+                if (t.getSize() > 1) {
+                    for (Player p : t.getMembers()) {
+                        for (Player p2 : t.getMembers()) {
+                            if (p2 == p || !p.getLocation().getWorld().equals(p2.getLocation().getWorld())) continue;
+                            if (distance == 0) {
+                                distance = (int) p.getLocation().distance(p2.getLocation());
+                            } else if ((int) p.getLocation().distance(p2.getLocation()) < distance) {
+                                distance = (int) p.getLocation().distance(p2.getLocation());
+                            }
                         }
+                        BedWars.nms.playAction(p, getMsg(p, Messages.FORMATTING_ACTION_BAR_TRACKING).replace("%bw_team%", t.getColor().chat() + t.getDisplayName(Language.getPlayerLanguage(p)))
+                                .replace("%bw_distance%", t.getColor().chat().toString() + distance).replace("&", "§"));
                     }
-                    BedWars.nms.playAction(p, getMsg(p, Messages.FORMATTING_ACTION_BAR_TRACKING).replace("%bw_team%", t.getColor().chat() + t.getDisplayName(Language.getPlayerLanguage(p)))
-                            .replace("%bw_distance%", t.getColor().chat().toString() + distance).replace("&", "§"));
                 }
             }
 
-        }
-
-        /* AFK SYSTEM FOR PLAYERS */
-        int current = 0;
-        for (Player p : getArena().getPlayers()) {
-            if (Arena.afkCheck.get(p.getUniqueId()) == null) {
-                Arena.afkCheck.put(p.getUniqueId(), current);
-            } else {
-                current = Arena.afkCheck.get(p.getUniqueId());
-                current++;
-                Arena.afkCheck.replace(p.getUniqueId(), current);
-                if (current == 45) {
-                    BedWars.getAPI().getAFKUtil().setPlayerAFK(p, true);
+            /* AFK SYSTEM FOR PLAYERS */
+            int current = 0;
+            for (Player p : getArena().getPlayers()) {
+                if (Arena.afkCheck.get(p.getUniqueId()) == null) {
+                    Arena.afkCheck.put(p.getUniqueId(), current);
+                } else {
+                    current = Arena.afkCheck.get(p.getUniqueId());
+                    current++;
+                    Arena.afkCheck.replace(p.getUniqueId(), current);
+                    if (current == 45) {
+                        BedWars.getAPI().getAFKUtil().setPlayerAFK(p, true);
+                    }
```

Comment:
It looks like you just put the afk system inside the for loop

Reply:
> It looks like you just put the afk system inside the for loop

Fixed